snapcraft/internal/build_providers/_base_provider.py

```diff
@@ -203,9 +203,18 @@ def _setup_snapcraft(self) -> None:
             snap_dir_unmounter=self._unmount_snaps_directory,
             file_pusher=self._push_file,
         )
+        # Inject snapcraft
         snap_injector.add(snap_name="core")
         snap_injector.add(snap_name="snapcraft")
 
+        # Also inject the base when confinement is set to classic to have real
+        # paths to the linker.
```

Comment:
This is part of the environment used by snapcraft. If we inject snapcraft and don't inject the base, different libraries could potentially be used, no? Is there a downside to always injecting this?

Reply:
A constant download if not installed locally.

Reply:
we could revisit this if it is direly needed, but so far have not made use of it